### PR TITLE
feat(pagination): add disabled pagination button styles

### LINKED ISSUE #1693: Pagination button styles should reflect disabled state
When the user reaches the end of the navigable page range, the forward pagination button is disabled and should be styled accordingly. Similarly, the backward pagination button should be visually reflect the disabled state at the start of the page range

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1693

This PR adds disabled pagination button styles

### CODE REVIEW

File: src/components/pagination/_pagination.scss [hunk 1]
```diff
@@ -318,6 +318,11 @@ $css--helpers: true;
         fill: $ui-05;
       }
     }
+
+    &--no-index {
```
Comment (joshblack) [inline src/components/pagination/_pagination.scss]: Should we un-nest the selector here?

**Review by joshblack — COMMENTED** [review]
Is there a good spec to reference for this variant? Also, is there an easy way to preview it in the netlify environment?

**Comment by emyarod** [comment]
from what I understand, we typically leave it up to the consumer of the vanilla repo to implement component logic like this. Since this style would only be visible when pagination is functional, not sure what the best way to provide a reference is aside from implementing a fully functional component in the docs

cc @asudoh

but I suppose in the meantime a quick way to preview this in netlify would be to manually add the `.bx--pagination__button--no-index` class to the pagination buttons

File: src/components/pagination/_pagination.scss (hunk 1 above)
Comment (emyarod) [reply src/components/pagination/_pagination.scss]: do we have guidelines on when to nest or when not to nest styles? I was just making use of SCSS language feature but wasn't aware that we intentionally avoid that or anything

**Comment by asudoh** [comment]
@emyarod Would it make sense to add a Fractal variant demonstrating the `--no-index` style...? Thanks!

**Comment by aagonzales** [comment]
Could a dev just paste a screen shot for review in the thread if its too hard to trigger?

**Comment by emyarod** [comment]
netlify should have a new component example now

File: src/components/pagination/pagination.config.js [hunk 2]
```diff
@@ -79,16 +80,29 @@ const variants = [
...
+    notes: `Notify the user of their position in the page range by disabling the appropriate pagination buttons at the start or end of the range.`,
+  },
 ];
 
 module.exports = {
   context: {
     featureFlags,
     prefix,
-    pageNumberChoices,
+    // pageNumberChoices,
```
Comment (asudoh) [inline src/components/pagination/pagination.config.js]: Are we removing this line...?

**Review by asudoh — APPROVED** [review]
LGTM 👍, just one thing - Thanks @emyarod for the updates!

File: src/components/pagination/pagination.config.js (hunk 2 above)
Comment (emyarod) [reply src/components/pagination/pagination.config.js]: yes I've removed it, left it in by accident

**Review by aagonzales — COMMENTED** [review]
I can't tell what color the carets are but they should be `$disabled-02` `#bebebe`. They look a little darker, just want to double check.

**Comment by emyarod** [comment]
thanks for the correction, the disabled icons should be `#bebebe` now

**Review by aagonzales — APPROVED** [review]

**Pull request merged**